include/modem/location.h

```diff
@@ -379,9 +442,24 @@ int location_agps_data_process(const char *buf, size_t buf_len);
  *
  * @return 0 on success, or negative error code on failure.
  * @retval -EINVAL Given buffer is NULL or buffer length zero.
+ * @retval -ENOTSUP CONFIG_LOCATION_GNSS_PGPS_EXTERNAL is not set.
  */
 int location_pgps_data_process(const char *buf, size_t buf_len);
 
+/**
+ * @brief Pass cellular location back to library.
+ *
+ * @details If Location library is not receiving cellular position directly from services,
+ * it triggers the LOCATION_EVT_CELLULAR_REQUEST event when neighbor cell information
+ * should be sent to cloud services for location resolution. Once application has obtained
+ * the location it can be handed over to Location library using this function.
+ *
+ * @param[in] location Cellular location results. NULL indicates that location
+ *                     couldn't be retrieved.
+ * @param[in] undefined Valid if location parameter is NULL. Defines whether
+ *                      cellular positioning result is unknown or failure.
+ */
+void location_cellular_ready(struct location_data *location, bool undefined);
```

Comment:
I find this API a bit confusing. Sorry for nitpicking...

I feel that "validity" is more appropriate for an output parameter. You always need to set the parameter when calling the function to either true or false, so the parameter is as such "valid". Maybe you could say that the parameter is only used when `location` parameter is NULL.

The parameter name is `undefined`, but it "defines whether cellular positioning result is **unknown** or **failure**". It would be good to use the same term, I think. The whole "undefined" term (here and in the event name) doesn't sound very nice to me, but I don't know what else we should call it...

And finally the `location` parameter description says that NULL indicates that the location couldn't be retrieved. In some cases location could have been retrieved, but we just don't need it sent back to the device. 😄

Reply:
For me, it felt strange that there isn't _ext_ word included in function name, but seems that there isn't _EXT_ in event names either. Only Kconfigs have those.
But could the name be alligned with the triggering event name or aligned with agps/pgps functions?

Reply:
What if this would be turned the other way around, for example:
`void location_cellular_ready(bool success, struct location_data *location);`
`success` would be set to `true` unless requesting the location failed and `location` would contain a valid location if one is available.

Reply:
> What if this would be turned the other way around, for example: `void location_cellular_ready(bool success, struct location_data *location);` `success` would be set to `true` unless requesting the location failed and `location` would contain a valid location if one is available.

That wouldn't behave the way the function does at the moment. And bool success don't provide anything useful compared to non-NULL vs NULL pointer. There are 3 states of outcome which the current parameters are able to tell while your proposal has two outcomes if I didn't get it wrong:

1. Success and location returned: location != NULL, undefined ignored
2. Failure: location = NULL, undefined = false
3. Undefined/unknown: location = NULL, undefined = true

Reply:
1. Success: success = true, location = <valid_ptr>
2. Failure: success = false, location = NULL
3. Undefined/unknown: success = true, location = NULL

This is what I was after. The 3rd case is success because no error happened (that we know of).

Reply:
Based on offline discussion, this function has been renamed to `location_cellular_ext_result_set()` and parameters modified so please review the new version.